### ISSUE TITLE
JNADominoProcess throws exceptions on multiple runs in the same process even when jnx.noterm is set

### ISSUE THREAD

**Issue by jesse-gallagher** [HCL-TECH-SOFTWARE/domino-jnx#291]
Specifically, this seems to be caused by `processTerminated` being set. In a case where the `jnx.noterm` property is configured, that should instead be taken to mean "JNX is not in charge of the Notes process and should not assume so", and so it should be safe to call `DominoProcess.get().terminateProcess()` multiple times.

Alternatively, it may be best to fix this overall by making it so that a given property or environment detection ( https://github.com/HCL-TECH-SOFTWARE/domino-jnx/issues/20 ) makes it so that `DominoProcess` init/term doesn't need to be called at all.

![Screen Shot 2021-12-14 at 10 12 28 AM](https://user-images.githubusercontent.com/935700/146031416-1a172f23-aa6c-4d49-a228-5cb6632203fc.png)

**Comment by RonaldAtGitHub** [comment]
Is there a way to call DominoProcess.get().initializeProcess(new String[0]) + DominoProcess.get().terminateProcess() multiple times, without this bug being solved? We are calling domino from api endpoints and want to initialize and stop the process with every call.

**Comment by klehmann** [comment]
I am not sure if calling initialize/terminateProcess multiple times is supported by Domino. I remember that we had issues doing this and the recommended way was to call them once when an application that accesses Domino starts and shuts down and then just call initialize/termThread for the current thread, which BTW fall back to the classic lotus.domino.NotesThread.sinitThread()/.stermThread() where possible since these internally contain more initialization code than their C API equivalents.
@jesse-gallagher what do you think?

**Comment by RonaldAtGitHub** [comment]
Thanks for the response.

We build our code in Java and use notes.jar in conjunction with recycle() method to interact with Domino. We build an update site for clients and use pde tooling while developing. In our code we recycle as soon as possible and that results in very stable code. 

Now we started to use domino-jnx and only call the terminateProcess if we shut down the HTTP (via the osgi BundleActivator stop method). 

While developing, a lot of our developers just shut down the server or switch branches during development without restarting the domino server, resulting in not calling the terminateProcess. Doing so crashes the server on the next start-up. 

I'm trying to find a solution for them.

**Comment by klehmann** [comment]
I don't think you have to call initializeProcess/terminateProcess at all if your code is running within the HTTP task. Do we enforce this in the API?

**Comment by jesse-gallagher** [comment]
Yeah, ideally those methods would never be called at all inside a task like nhttp, but the API enforces that internal flags have been set. The trouble is that we can't know for sure that it was already initialized without initializing it. I'm considering now something like an OSGi fragment bundle in the XSP distribution that tips off the API that those methods should either not be called or should be no-ops.

Probably it'll be better if such a fragment causes JNX to skip the "has it been initialized?" test for both the process and the thread - since the process should never be init/termed, but a thread _might_ need to be.
